src/components/Models/EnvironmentModel/EnvironmentModel.component.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { FC } from 'react';
+import { useFallback, useGltfLoader, CustomNode, Transform } from 'src/services';
+import { useGraph } from '@react-three/fiber';
+import { BaseModelProps } from 'src/types';
+import { EnvironmentModels } from 'src/services/Environment.service';
+
+export interface EnvironmentModelProps extends BaseModelProps {
+  environment: string | EnvironmentModels;
+  scale?: number;
+}
+
+export const EnvironmentModel: FC<EnvironmentModelProps> = ({ environment, scale = 1, setModelFallback }) => {
+  const transform = new Transform();
+  const { scene } = useGltfLoader(environment);
+  const { nodes } = useGraph(scene);
+
+  useFallback(nodes, setModelFallback);
+
+  return (
+    <group>
+      {Object.keys(nodes).map((key) => {
+        const node = nodes[key] as CustomNode;
+        if (node.type === 'Mesh') {
+          return (
+            <mesh
+              receiveShadow
+              key={node.name}
+              scale={scale}
+              position={transform.position}
+              rotation={transform.rotation}
+              geometry={node.geometry}
+              material={node.material}
+              morphTargetInfluences={node.morphTargetInfluences || []}
+            />
+          );
+        }
+
+        return null;
```

Comment:
not a big deal but maybe we can do early return

Reply:
I don't think so, it might be the case where `node.type` is `SkinnedMesh` and that would not work with `<mesh />`.
Would have to guard against every other type that isn't a mesh.